src/datatypes2/src/vectors/string.rs

```diff
@@ -0,0 +1,262 @@
+    }
+
+    fn null_count(&self) -> usize {
+        self.array.null_count()
+    }
+
+    fn is_null(&self, row: usize) -> bool {
+        self.array.is_null(row)
+    }
+
+    fn slice(&self, offset: usize, length: usize) -> VectorRef {
+        let data = self.array.data().slice(offset, length);
+        Arc::new(Self::from_array_data(data))
+    }
+
+    fn get(&self, index: usize) -> Value {
+        vectors::impl_get_for_vector!(self.array, index)
+    }
+
+    fn get_ref(&self, index: usize) -> ValueRef {
+        vectors::impl_get_ref_for_vector!(self.array, index)
+    }
+}
+
+impl ScalarVector for StringVector {
+    type OwnedItem = String;
+    type RefItem<'a> = &'a str;
+    type Iter<'a> = ArrayIter<&'a StringArray>;
+    type Builder = StringVectorBuilder;
+
+    fn get_data(&self, idx: usize) -> Option<Self::RefItem<'_>> {
+        if self.array.is_valid(idx) {
+            Some(self.array.value(idx))
+        } else {
+            None
+        }
+    }
+
+    fn iter_data(&self) -> Self::Iter<'_> {
+        self.array.iter()
+    }
+}
+
+pub struct StringVectorBuilder {
+    mutable_array: MutableStringArray,
+}
+
+impl MutableVector for StringVectorBuilder {
+    fn data_type(&self) -> ConcreteDataType {
+        ConcreteDataType::string_datatype()
+    }
+
+    fn len(&self) -> usize {
+        self.mutable_array.len()
+    }
+
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn as_mut_any(&mut self) -> &mut dyn Any {
+        self
+    }
+
+    fn to_vector(&mut self) -> VectorRef {
+        Arc::new(self.finish())
+    }
+
+    fn push_value_ref(&mut self, value: ValueRef) -> Result<()> {
+        match value.as_string()? {
+            Some(v) => self.mutable_array.append_value(v),
+            None => self.mutable_array.append_null(),
+        }
+        Ok(())
+    }
+
+    fn extend_slice_of(&mut self, vector: &dyn Vector, offset: usize, length: usize) -> Result<()> {
+        vectors::impl_extend_for_builder!(self, vector, StringVector, offset, length)
+    }
+}
+
+impl ScalarVectorBuilder for StringVectorBuilder {
+    type VectorType = StringVector;
+
+    fn with_capacity(capacity: usize) -> Self {
+        Self {
+            mutable_array: MutableStringArray::with_capacity(capacity, 0),
+        }
+    }
+
+    fn push(&mut self, value: Option<<Self::VectorType as ScalarVector>::RefItem<'_>>) {
+        match value {
+            Some(v) => self.mutable_array.append_value(v),
+            None => self.mutable_array.append_null(),
+        }
+    }
+
+    fn finish(&mut self) -> Self::VectorType {
+        StringVector {
+            array: self.mutable_array.finish(),
+        }
+    }
+}
+
+impl Serializable for StringVector {
+    fn serialize_to_json(&self) -> Result<Vec<serde_json::Value>> {
+        self.iter_data()
+            .map(|v| match v {
+                None => Ok(serde_json::Value::Null),
+                Some(vec) => serde_json::to_value(vec),
+            })
```

Comment:
Could we use `serde_json::to_value()` directly? Maybe we need to add a test for this. BinaryVector should also use a similar pattern.
```suggestion
            .map(serde_json::to_value)
```